main.go

```diff
@@ -387,6 +390,48 @@ func main() {
 
 	// Start command execution
 	cmd.HandleFunc(ctx, args)
+
+	// Go routine for checking the GPS
+	go func() {
+		if fOptions.UpdateGrid {
+			for range time.Tick(time.Minute * 5) {
+				updateGridIfNeeded()
+			}
+		}
+	}()
+}
+
+// Update Grid if needed
+func updateGridIfNeeded() {
+	// create connection to gpsd
+	conn, err := gpsd.Dial(config.GPSd.Addr)
+	if err != nil {
+		log.Printf("GPSd daemon: %s", err)
+		return
+	}
+	defer conn.Close()
+	conn.Watch(true)
+
+	// get next position from gpsd
+	pos, err := conn.NextPos()
+	if err != nil {
+		log.Printf("GPSd: %s", err)
+		return
+	}
+
+	// get grid square from position
+	currGridSquare := pos.GridSquare
+
+	if currGridSquare != config.Locator {
+		// update config
+		config.Locator = currGridSquare
+		// write config
+		if err := WriteConfig(config, fOptions.ConfigPath); err != nil {
+			log.Printf("Unable to write config: %s", err)
+		}
```

Comment:
I don't think rewriting the actual config file is the best idea. It is not a lossless operation, as the user might have defined keys and values that are not part of the JSON spec (e.g. "comments"). Custom formatting will also be lost, and possibly also ordering of aliases etc. It also has the potential of corrupting the file if we for some reason the operation fails in the middle of writing (full disk, power loss++).

I guess it's useful to retain the "last known" position when restarting the app. Otherwise, the user would have to wait the 5 min interval before the locator field is updated again

As I see it, we have two options:
1. Persist the "last known position" in a separate JSON file, and use that as a initial value (if present).
2. Block on startup until we get a fresh position from GPSd.

I think the latter might be a better solution, as it guarantees running with `--locator-from-gpsd` uses a value from GPSd at all times.

Reply:
Thanks Martin...I agree the latter would be the better option.

Reply:
After some refreshing on this...won't the gaps.Dial() and conn.NextPos() already blocking?